packages/remix-react/components.tsx

```diff
@@ -1350,7 +1350,7 @@ type SerializeType<T> = T extends JsonPrimitives
         ? null
         : SerializeType<T[k]>;
     }
-  : T extends (infer U)[]
+  : T extends ReadonlyArray<infer U>
```

Comment:
Does this allow regular arrays as well?

Reply:
yes, because `Array<unknown>` extends `ReadonlyArray<unknown>`

Reply:
@JNaftali Can you point us to the docs of this please?
Been looking into it, but can't find that.

Maybe @mattpocock can also help out on this one?